### ISSUE TITLE
Internal compiler error

### ISSUE THREAD

**Issue by vporton** [ethereum/solidity#9988]
## Description

Internal compiler error of Solidity 0.7.1.

## Environment

- Compiler version: 0.7.1
- Target EVM version (as per compiler settings): default
- Framework/IDE (e.g. Truffle or Remix): Buidler
- Operating system: Ubuntu 20.04.1

## Steps to Reproduce

Download https://github.com/vporton/eth-tokens-sum/tree/solidity0.7.1-internal-error

```
$ npx buidler compile --show-stack-traces
Compiling...
InternalCompilerError: I sense a disturbance in the stack: 5 vs 6

Error BDLR600: Compilation failed

BuidlerError: BDLR600: Compilation failed
    at SimpleTaskDefinition.action (/home/porton/Projects/bounties/eth-tokens-sum/node_modules/@nomiclabs/buidler/src/builtin-tasks/compile.ts:160:13)
    at Environment._runTaskDefinition (/home/porton/Projects/bounties/eth-tokens-sum/node_modules/@nomiclabs/buidler/src/internal/core/runtime-environment.ts:203:14)
    at SimpleTaskDefinition.action (/home/porton/Projects/bounties/eth-tokens-sum/node_modules/@nomiclabs/buidler/src/builtin-tasks/compile.ts:203:31)
    at Environment._runTaskDefinition (/home/porton/Projects/bounties/eth-tokens-sum/node_modules/@nomiclabs/buidler/src/internal/core/runtime-environment.ts:203:14)
    at Environment._runTaskDefinition (/home/porton/Projects/bounties/eth-tokens-sum/node_modules/@nomiclabs/buidler/src/internal/core/runtime-environment.ts:203:14)
    at main (/home/porton/Projects/bounties/eth-tokens-sum/node_modules/@nomiclabs/buidler/src/internal/cli/cli.ts:157:5)
```

I will work on creating a minimal reproducible example.

**Comment by vporton** [comment]
Minimal reproducible example:
```solidity
//SPDX-License-Identifier: Apache-2.0	
pragma solidity ^0.7.1;

contract TokensFlow
{
    function exchangeToParent(uint256 _id) external {
        _doBurn("");
    }

    function _doBurn(bytes calldata _data) internal {
    }
}
```

**Comment by bshastry** [comment]
Thank you for reporting the issue @vporton . I'm not able to reproduce this on the latest release `0.7.3`. Does the issue reproduce with `0.7.2`?

**Comment by chriseth** [comment]
Fixed in 0.7.2.
